tests/ui/lint/implicit_autorefs.stderr

```diff
@@ -0,0 +1,111 @@
+warning: implicit auto-ref creates a reference to a dereference of a raw pointer
+  --> $DIR/implicit_autorefs.rs:12:14
+   |
+LL |     addr_of!((*ptr)[..16])
+   |              ^^^^^^^^^^^^
+   |
+   = note: creating a reference requires the pointer to be valid and imposes aliasing requirements
+   = note: `#[warn(dangerous_implicit_autorefs)]` on by default
+help: try using a raw pointer method instead; or if this reference is intentional, make it explicit
+   |
+LL |     addr_of!((&(*ptr))[..16])
+   |              ++      +
```

Comment:
While catching up with the comments posted in the now closed PR, I thought the plan was to avoid suggesting explicit borrows and to instead suggest `#[allow]`'ing the lint: Sth. like *if this reference is intentional to make it explicit by adding `#[allow(implicit_unsafe_autorefs)]`*.

CC 1st paragraph of https://github.com/rust-lang/rust/pull/103735#issuecomment-1370420305; https://github.com/rust-lang/rust/pull/103735#issuecomment-1826826298; https://github.com/rust-lang/rust/pull/103735#issuecomment-1826886519

Reply:
There has been concern from T-lang members about only having `#[allow]` as an option for lints, cc https://github.com/rust-lang/rust/pull/118833#issuecomment-1875723905 (and maybe a bit https://github.com/rust-lang/rust/issues/122759#issue-2196513165).

I think it would make sense to ask T-lang that question in the nomination to see what they want, worst case it's just removing some code.

Reply:
FWIW, nothing in #122759 is meant to imply that we shouldn't have lints whose suggestion is `#[allow]`.

Reply:
Good to know. Thanks.

Reply:
> I think it would make sense to ask T-lang about this in the nomination, to see what they want, worst case it's just removing some code.

@Urgau how do you want to proceed here, should we ask for T-lang feedback?

Reply:
We want to nominate for T-lang in any case -- once I've reviewed this PR ^^' I'll really try to get to it tonight.

Reply:
Hi @fmease, just wanted to remind you to take a look at this PR if you are still interested. Thanks! 🙂